weaviate/collections/data.py

```diff
@@ -480,7 +471,9 @@ def update(
 
         self._update(weaviate_obj, uuid=uuid)
 
-    def reference_add(self, from_uuid: UUID, from_property: str, to: WeaviateReference) -> None:
+    def reference_add(
+        self, from_uuid: UUID, from_property: str, to: Union[UUID, _Reference]
```

Comment:
should we have an extra type for this?

We could have `ReferenceInputWithList` and `ReferenceInputSingleObject` (not a big fan of the name)